sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubAsyncClient.java

```diff
@@ -70,29 +71,31 @@
     private static final String SENDER_ENTITY_PATH_FORMAT = "%s/Partitions/%s";
 
     private final ClientLogger logger = new ClientLogger(EventHubAsyncClient.class);
+    private final MessageSerializer messageSerializer;
     private final Mono<EventHubConnection> connectionMono;
     private final AtomicBoolean hasConnection = new AtomicBoolean(false);
-    private final ConnectionOptions connectionOptions;
+    private final ConnectionOptions options;
     private final String eventHubName;
     private final EventHubProducerOptions defaultProducerOptions;
     private final EventHubConsumerOptions defaultConsumerOptions;
     private final TracerProvider tracerProvider;
 
-    EventHubAsyncClient(ConnectionOptions connectionOptions, TracerProvider tracerProvider,
+    EventHubAsyncClient(ConnectionOptions options, TracerProvider tracerProvider, MessageSerializer messageSerializer,
                         Mono<EventHubConnection> eventHubConnectionMono) {
 
-        this.connectionOptions = Objects.requireNonNull(connectionOptions, "'connectionOptions' cannot be null.");
+        this.options = Objects.requireNonNull(options, "'connectionOptions' cannot be null.");
```

Comment:
The error message should match the variable name.  `"'options' cannot be null."`

Also, I liked the previous name -  `connectionOptions`